os/drivers/audio/alc5658.c

```diff
@@ -570,6 +586,19 @@ static int alc5658_getcaps(FAR struct audio_lowerhalf_s *dev, int type, FAR stru
 			 */
 		}
 
+		switch (caps->ac_format.hw) {
```

Comment:
For query, we use caps->ac_subtype for inner case and get values  save into ac_controls.
For setting, we use caps->ac_format.hw for inner case and set value from ac_controls.
So here we should use ac_subtype.